### PR TITLE
Add Budget Function and Subfunction filters to Account Data Download

### PR DESCRIPTION
**Pull request by kojodoesdesign**
**High level description:**
Add Budget Function and Subfunction filters to Account Data Download page.

**Technical details:**
- Adds Budget Function into API contract
- Adds filter message with a readiness status based on Budget Function/Account
- Allows download button to be highlighted if Budget Function/Account are selected

**Link to JIRA Ticket:**
[DEV-1446](https://federal-spending-transparency.atlassian.net/browse/DEV-1446)

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] Design Review
- [x] Contract Approved by backend  (@michaeldhess)

### CODE REVIEW

**Comment by gitapablabah** [comment]
New top section looks great. Update the bottom section 'Your selected options are...' to match the order and labeling of the mock up. TY.

**Comment by gitapablabah** [comment]
UI is good to go. TY!

File: api_contracts/contracts/downloadCenter/DownloadCenter.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ HOST: https://api.usaspending.gov
 
 # Download Center
 
-These endpoints are used to power USAspending.gov's download center. 
+These endpoints are used to power USAspending.gov's download center. This data can be used to create account data files.
```
Comment (ebdabbs) [inline api_contracts/contracts/downloadCenter/DownloadCenter.md]: `This data can be used to create account data files.` can be removed (might be leftover from a previous version of this file). We will eventually include custom award data download and data dictionary endpoints and will want to leave the top-level description more general.

File: api_contracts/contracts/downloadCenter/DownloadCenter.md [hunk 2]
```diff
@@ -92,19 +124,31 @@ This endpoint returns a list of budget subfunctions with their associated title
...
-## BudgetFunctionResults (object)
-+ results (required, array[BudgetFunction], fixed-type)
+## AgencyData
++ cfo_agencies: (required, array[AgencyCFO])
++ other_agencies: (required, array[AgencyOtherAgencies])
++ federal_accounts: (required, array)
++ sub_agencies: (required, array)
+
+## AgencyCFO
++ name: `Department of Agriculture`
```
Comment (ebdabbs) [inline api_contracts/contracts/downloadCenter/DownloadCenter.md]: Please include `optional` or `required` and data type for the data structure attributes

File: api_contracts/contracts/downloadCenter/DownloadCenter.md [hunk 3]
```diff
@@ -53,7 +57,33 @@ This endpoint returns a list of budget subfunctions with their associated title
 
 + Response 200 (application/json)
     + Attributes
-        + results (BudgetSubfunctionResults, fixed-type) 
+        + results (required, array[BudgetSubfunction], fixed-type)
+
+## Agency Function [/api/v2/bulk_download/list_agencies/]
```
Comment (ebdabbs) [inline api_contracts/contracts/downloadCenter/DownloadCenter.md]: Not sure why these have `Function` in the name - something like `List Agencies`, `Federal Accounts by Agency` may be more clear

File: api_contracts/contracts/downloadCenter/DownloadCenter.md [hunk 4]
```diff
@@ -38,9 +38,13 @@ This endpoint returns a list of budget functions with their associated title and
 
 ### Budget Function [GET]
 
++ Parameters
+    + agency: `all` (optional, string)
+    + federal_account: `212` (optional, string)
```
Comment (michaeldhess) [inline api_contracts/contracts/downloadCenter/DownloadCenter.md]: These parameters have not yet been implemented in the `/list_budget_functions/` endpoint, and will not be implemented until we migrate to using the Federal Hierarchy data for our top- and sub-tier agency data.

File: src/js/components/bulkDownload/accounts/filters/BudgetFilter.jsx [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+        this.setState({
+            showBudgetFunctionPicker: !this.state.showBudgetFunctionPicker,
+            showBudgetSubfunctionPicker: false
+        });
+    }
+
+    toggleBudgetSubfunctionPicker(e) {
+        e.preventDefault();
+        this.setState({
+            showBudgetSubfunctionPicker: !this.state.showFederalAccountPicker,
```
Comment (ebdabbs) [inline src/js/components/bulkDownload/accounts/filters/BudgetFilter.jsx]: `!this.state.showFederalAccountPicker` should be `!this.state.showBudgetSubfunctionPicker`

File: src/js/components/bulkDownload/accounts/filters/BudgetFilter.jsx [hunk 6]
```diff
@@ -0,0 +1,264 @@
+/**
+ * BudgetFilter.jsx
```
Comment (ebdabbs) [inline src/js/components/bulkDownload/accounts/filters/BudgetFilter.jsx]: Can we rename this to `BudgetFunctionFilter`?

File: src/js/components/bulkDownload/accounts/filters/AgencyFilter.jsx [hunk 7]
```diff
@@ -174,112 +170,115 @@ export default class AgencyFilter extends React.Component {
...
-                                <ul>
-                                    <li className="field-item">
-                                        <button
-                                            className="item-button group-label"
-                                            title="Federal Accounts"
-                                            aria-label="Federal Accounts"
-                                            disabled>
+                                <div className={`field-list ${showFederalAccountPicker}`}>
+                                    <ul>
+                                        <li className="field-item">
```
Comment (ebdabbs) [inline src/js/components/bulkDownload/accounts/filters/AgencyFilter.jsx]: I don't think this first disabled option is necessary. (In the agency dropdown we use something similar to separate CFO agencies and Other Agencies)

File: src/js/components/bulkDownload/accounts/filters/BudgetFunctionFilter.jsx [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+                                                name="All"
+                                                value="all"
+                                                onClick={this.handleBudgetFunctionSelect}>
+                                            All
+                                            </button>
+                                        </li>
+                                        <li className="field-item">
+                                            <button
+                                                className="item-button group-label"
+                                                title="CFO Agencies"
```
Comment (ebdabbs) [inline src/js/components/bulkDownload/accounts/filters/BudgetFunctionFilter.jsx]: Same as previous comment. We can remove the budget sub-functions one as well.

**Pull request merged**